pkg/secrets/secrets.go

```diff
@@ -99,7 +99,65 @@ func generateUserDataWithPubKey(pubKey string) string {
 			$acl.SetAccessRule($administratorsRule)
 			$acl.SetAccessRule($systemRule)
 			$acl | Set-Acl
-			Restart-Service sshd`
+			Restart-Service sshd
+			`
+
+	if platformType == oconfig.AWSPlatformType {
+		userData = appendAwsUserDataConfig(userData)
+	}
+	return userData
+}
+
+// appendAwsUserDataConfig appends the AWS EC2Launch installation script to
+// the given userData to persist the instance metadata routes
+func appendAwsUserDataConfig(userData string) string {
+	// EC2Launch v2 minimum version where the task to persist routes was introduced
+	// see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2launchv2-versions.html#ec2launchv2-version-history
+	EC2LaunchMinimumVersion := "2.0.1643"
+
+	// S3 URL for the latest EC2Launch version
+	// see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2launch-v2-install.html#lv2-download-s3
+	EC2LaunchS3LatestURL := "https://s3.amazonaws.com/amazon-ec2launch-v2/windows/amd64/latest/AmazonEC2Launch.msi"
+
+	// append the EC2Launch installation script
+	userData += `function Install-LatestEC2LaunchV2 {
+				# set download dir
+				$DownloadDir = "$env:USERPROFILE\Desktop\EC2Launchv2"
+				New-Item -Path "$DownloadDir" -ItemType Directory -Force
+				# URL for your download location
+				$Url = "` + EC2LaunchS3LatestURL + `"
+				# set download location
+				$DownloadFile = "$DownloadDir" + "\" + $(Split-Path -Path $Url -Leaf)
+				# download the agent
+				Invoke-WebRequest -Uri $Url -OutFile $DownloadFile
+				# run the install
+				msiexec /i "$DownloadFile"
+			}
+			$EC2LaunchMinimumVersion = "` + EC2LaunchMinimumVersion + `"
+			$EC2LaunchExeLocation = "$env:ProgramFiles\Amazon\EC2Launch\EC2Launch.exe"
+			
+			if ( -not (Test-Path -Path $EC2LaunchExeLocation -PathType Leaf) ) {
+				Write-Output "EC2Launch binary not found in $EC2LaunchExeLocation, installing..."
+			
+				Install-LatestEC2LaunchV2
+			}
+			
+			$currentVersion = $(& $EC2LaunchExeLocation version)
+			
+			# check supported minimum version
+			if ($currentVersion -lt $EC2LaunchMinimumVersion) {   
+				Write-Output "EC2Launch upgrading from version $currentVersion"
+			
+				Install-LatestEC2LaunchV2
+			
+				$currentVersion = $(& $EC2LaunchExeLocation version)
+			}
+			
+			Write-Output "EC2Launch version $currentVersion"
```

Comment:
Ran a test job as a validation of this logic: 
See `EC2Launch version 2.0.1948` in 


https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_windows-machine-config-operator/2296/pull-ci-openshift-windows-machine-config-operator-master-aws-e2e-operator/1821143236547186688/artifacts/aws-e2e-operator/windows-e2e-operator-test/artifacts/pods/job-name=print-logs-aws-ec2-userdata-10-0-57-26-job-qwflf/job-name=print-logs-aws-ec2-userdata-10-0-57-26-job-qwflf.log